### PR TITLE
Copyright headers

### LINKED ISSUE #93: Add license headers for C# source files
## Why 

https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#code-attribution

## What

1. License information should be included in all source files where applicable.
2. Add a CI check to validate if all (applicable) source files have license information (at least C# and C++ source code)
3. Remove from `src/Datadog.Trace/GlobalSuppressions.cs`:
```
// Copyright claims are suppressed until re-naming is finalized
[assembly: SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1636:File header copyright text must match", Justification = "Reviewed.")]
[assembly: SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1641:File header company name text must match", Justification = "Reviewed.")]
[assembly: SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1638:File header file name documentation must match file name", Justification = "Reviewed.")]
```

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #93 

Changes proposed in this pull request:
Copyright headers for C# files.
It should be verified in CI. The only exception is Sample.AspNet application. As I know, it cannot be converted to SDK csproj format, so it is not easily supporting Stylecop.Analyzer.
A lot of small fixes, related to enabling stylecop for tests/samples apps.

Copyright text taken from https://github.com/open-telemetry/opentelemetry-dotnet/blob/d1c34ab235edc3816704c3a06fa8d3573efd9d9a/build/stylecop.json#L5-L6

### CODE REVIEW

**Review by pellared — APPROVED** [review]
Shouldn't we do the same for native C++ code? 

Another issue and PR if so. For reference, I guess there may be some automation here: https://github.com/open-telemetry/opentelemetry-cpp. Personally, I am aware of https://github.com/open-telemetry/opentelemetry-go/blob/b1c1e781e6df6ac8a280da21766a00d16a45a9ed/Makefile#L171-L179

File: OpenTelemetry.AutoInstrumentation.sln [hunk 1]
```diff
@@ -112,11 +109,8 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Samples.MongoDB", "test\test-applications\integrations\Samples.MongoDB\Samples.MongoDB.csproj", "{02D6A472-A602-4BF4-A539-8AD165097898}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "aspnet", "aspnet", "{2E1F47D9-BEC1-4DCE-A84D-847FE5BBBE9E}"
-	ProjectSection(SolutionItems) = preProject
```
Comment (pjanotti) [inline OpenTelemetry.AutoInstrumentation.sln:115]: 👍

File: samples/ConsoleApp.SelfBootstrap/Program.cs [hunk 2]
```diff
@@ -28,7 +44,7 @@ private static async Task<int> Main()
             });
 
         _tracerProvider = builder.Build();
-
```
Comment (pjanotti) [inline samples/ConsoleApp.SelfBootstrap/Program.cs:31]: Q.: don't we have a StyleCop to enforce no whitespace lines? Or it wasn't being enforced for this file?

File: src/OpenTelemetry.AutoInstrumentation/Util/PropertyFetcher.cs [hunk 3]
```diff
@@ -1,4 +1,18 @@
-// From https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Util/PropertyFetcher.cs:1]: This comment seems worth to preserve below the header.

File: test/integration-tests/IntegrationTests.Helpers/GlobalSuppressions.cs [hunk 4]
```diff
@@ -1,7 +1,18 @@
-// This file is used by Code Analysis to maintain SuppressMessage
```
Comment (pjanotti) [inline test/integration-tests/IntegrationTests.Helpers/GlobalSuppressions.cs:1]: Perhaps preserver this comment below the header.

File: test/test-applications/integrations/Samples.GraphQL/Program.cs [hunk 5]
```diff
@@ -19,8 +35,7 @@ public static void Main(string[] args)
             .UseKestrel(serverOptions =>
                 // Explicitly set AllowSynchronousIO to true since the default changes
                 // between AspNetCore 2.0 and 3.0
-                serverOptions.AllowSynchronousIO = true
-            )
+                serverOptions.AllowSynchronousIO = true)
```
Comment (pjanotti) [inline test/test-applications/integrations/Samples.GraphQL/Program.cs:38]: Q.: just curious, was this pointed by StyleCop?

**Review by pjanotti — APPROVED** [review]
LGTM. Just a couple of files that perhaps should preserve some comments that were removed, but it is fine to revisit them later.

**Comment by pjanotti** [comment]
@pellared @Kielek I created #418 so we can do the native sources in a separate PR.

File: test/test-applications/integrations/Samples.GraphQL/Program.cs (hunk 5 above)
Comment (Kielek) [reply test/test-applications/integrations/Samples.GraphQL/Program.cs:38]: Yes.

File: samples/ConsoleApp.SelfBootstrap/Program.cs (hunk 2 above)
Comment (Kielek) [reply samples/ConsoleApp.SelfBootstrap/Program.cs:31]: It was not enforced for sample folder. Now all projects should be covered by Stylecop.Analyzers.

File: test/integration-tests/IntegrationTests.Helpers/GlobalSuppressions.cs (hunk 4 above)
Comment (Kielek) [reply test/integration-tests/IntegrationTests.Helpers/GlobalSuppressions.cs:1]: Comment restored.

File: src/OpenTelemetry.AutoInstrumentation/Util/PropertyFetcher.cs (hunk 3 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Util/PropertyFetcher.cs:1]: 25234cb2e130d72103d1ff0f4330ce36d1f2f1bc

**Comment by Kielek** [comment]
@pjanotti, @pellared separate task for native code and other files seems reasonable :)
Thanks for creating it.

**Pull request merged**